vhdbuilder/packer/test-scan-and-cleanup.sh

```diff
@@ -46,11 +46,12 @@ else
   echo -e "\n\nSkipping tests for CVM 20.04"
 fi
 
-if [ "$OS_VERSION" != "18.04" ]; then
+if [ "$OS_VERSION" != "18.04" ] && [ "$OS_VERSION" != "V3" ]; then
   SCRIPT_ARRAY+=("./vhdbuilder/packer/vhd-scanning.sh")
 else
   # 18.04 VMs don't have access to new enough 'az' versions to be able to run the az commands in vhd-scanning-vm-exe.sh
-  echo -e "\n\nSkipping scanning for 18.04"
+  # Azurelinux 3.0 does not have an 'az' package - https://microsoft.visualstudio.com/OS/_workitems/edit/52283489
+  echo -e "\n\nSkipping scanning for 18.04/AzureLinux 3.0"
```

Comment:
non-actionable: I feel like we might consider refactoring our scanning to a container-based approach so we don't have to worry about things like this (we'll always have a viable container runtime setup (containerd) to run said container as well) 

I think is also how kube-bench does their stuff: https://github.com/aquasecurity/kube-bench

@jason1028kr @AlisonB319  any thoughts on this?

Reply:
FYI tthe reason for needing az is to extract the trivy scanning files from the VM. Its only used to upload the files from the vm to storage and then the pipeline pulls the files from storage so that they are pipeline artifacts. 

So not sure how a container approach would change things, especially with Jason's new changes and additions - but always open to try new things!

As an aside - how does AzureLinux3.0 not have az if the older versions do? @aadhar-agarwal you should take a look at https://github.com/Azure/AgentBaker/blob/0549f85a78e1fbf31e9f7162fb91599c0b191333/vhdbuilder/packer/vhd-scanning-exe-on-vm.sh#L38

I would bet there would be a way to install az

Reply:
Hey @AlisonB319, yeah the logic to install az cli in `vhd-scanning-exe-on-vm.sh` is not applicable at the moment, since 3.0 does not have an az cli package at the moment. That logic is not able to install an applicable azure cli package for 3.0

We have this [work item](https://microsoft.visualstudio.com/OS/_workitems/edit/52283489) for this. I believe we are still waiting to get an az cli package for 3.0 and we will have one in the future.

Reply:
> FYI tthe reason for needing az is to extract the trivy scanning files from the VM. Its only used to upload the files from the vm to storage and then the pipeline pulls the files from storage so that they are pipeline artifacts.
> 
> So not sure how a container approach would change things, especially with Jason's new changes and additions - but always open to try new things!
> 
> As an aside - how does AzureLinux3.0 not have az if the older versions do? @aadhar-agarwal you should take a look at
> 
> https://github.com/Azure/AgentBaker/blob/0549f85a78e1fbf31e9f7162fb91599c0b191333/vhdbuilder/packer/vhd-scanning-exe-on-vm.sh#L38
> 
> I would bet there would be a way to install az

right, I was just thinking we might be able to procure a reproducible scanning environment via a container image where we have all the tools we need (e.g. az cli), might be worth playing around with at some point